experiments/opencog/pattern_matcher_vqa/tbd_cog/tbd_helpers.py

```diff
@@ -0,0 +1,253 @@
+    lst = atomspace.add_link(types.ListLink, [variable, eval_link_sub])
+    return atomspace.add_link(types.EvaluationLink, [predicate, lst])
+
+
+def build_filter(atomspace, filter_type, filter_argument, exec_out_sub):
+    schema = atomspace.add_node(types.GroundedSchemaNode, "py:filter")
+
+    lst = atomspace.add_link(types.ListLink, [filter_type,
+                                              filter_argument,
+                                              exec_out_sub])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_relate(atomspace, relate_argument, exec_out_sub):
+    schema = atomspace.add_node(types.GroundedSchemaNode, "py:relate")
+    lst = atomspace.add_link(types.ListLink, [atomspace.add_node(types.ConceptNode, relate_argument),
+                                              exec_out_sub])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_intersect(atomspace, arg0, arg1):
+    schema =  atomspace.add_node(types.GroundedSchemaNode, "py:intersect")
+    lst = atomspace.add_link(types.ListLink, [arg0,
+                                              arg1])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_bind_link(atomspace, eval_link, inheritance):
+    """
+    Build bind link from ExecutionOutput or Evaluation links and inheritance links
+
+    :param atomspace: AtomSpace
+    :param eval_link: ExecutionOutput or EvaluationLink
+    :param inheritance: Set[InheritanceLink]
+    :return: BindLink
+    """
+    varlist = []
+    for inh in inheritance:
+        for atom in inh.get_out():
+            if atom.type == types.VariableNode:
+                varlist.append(atom)
+    variable_list = atomspace.add_link(types.VariableList, varlist)
+    list_link = atomspace.add_link(types.ListLink, varlist + [eval_link])
+    conj = atomspace.add_link(types.AndLink, [*inheritance, eval_link])
+    bind_link = atomspace.add_link(types.BindLink, [variable_list, conj, list_link])
+    return bind_link
+
+
+def build_scene(atomspace, scene):
+    schema =  atomspace.add_node(types.GroundedSchemaNode, "py:init_scene")
+    lst = atomspace.add_link(types.ListLink, [scene])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def build_same(atomspace, same_argument, exec_out_sub):
+    schema = atomspace.add_node(types.GroundedSchemaNode, "py:same")
+    lst = atomspace.add_link(types.ListLink, [atomspace.add_node(types.ConceptNode, same_argument),
+                                              exec_out_sub])
+    return atomspace.add_link(types.ExecutionOutputLink, [schema, lst])
+
+
+def return_prog(atomspace, commands, inheritance_set=None):
+    """
+    Convert tbd program to atomese
+
+    :param atomspace: Atomspace
+    :param commands: list
+    :param inheritance_set: set
+        Inhertiance links constructed during conversion
+    :return: Tuple[ExecutionOutputLink, list, set]
```

Comment:
It is not clear how second returned value ```list``` is used?

Reply:
It is used in recursive calls to generate nested atoms